src/index.ts

```diff
@@ -326,11 +330,9 @@ function decodeRow(row: QueryResultRow): Array<string | null> {
 }
 
 export function cast(field: Field, value: string | null): any {
-  if (value === '' || value == null) {
-    return value
-  }
```

Comment:
If we're being type aware, we can't just return an empty string or null for numbers, we _can_, but it's more correct to coerce them in their specific switch branches so they get a `0`.